language/move-vm/runtime/src/loader.rs

```diff
@@ -668,27 +674,32 @@ impl Loader {
         bundle_verified: &BTreeMap<ModuleId, CompiledModule>,
         data_store: &mut impl DataStore,
     ) -> VMResult<()> {
-        self.verify_module(module, bundle_verified, data_store, true)
+        self.verify_module(
+            module,
+            bundle_verified,
+            &mut BTreeMap::new(),
+            data_store,
+            true,
+        )
     }
 
     fn verify_module_expect_no_missing_dependencies(
         &self,
         module: &CompiledModule,
+        in_progress: &mut BTreeMap<ModuleId, CompiledModule>,
         data_store: &mut impl DataStore,
     ) -> VMResult<()> {
-        self.verify_module(module, &BTreeMap::new(), data_store, false)
+        self.verify_module(module, &BTreeMap::new(), in_progress, data_store, false)
     }
 
     fn verify_module(
```

Comment:
Now that this no longer actually runs the verifier on `module`, maybe we should call it something like `check_dependencies` or `verify_dependencies`.

Overall, I think it might help us to pick some consistent terminology for the different things this component is doing (loading, verifying, linking, etc.) and refactor accordingly. A newtype state machine (e.g., with states for `LinkedModule`, `VerifiedModule`, etc.) might help us ensure that every module has passed through the expected steps exactly once.